text/0000-template-comments.md

```diff
@@ -0,0 +1,90 @@
+---
+title: HTML comments for LWC templates
+status: DRAFTED
+created_at: 2021-03-31
+updated_at: 2021-03-31
+pr: https://github.com/salesforce/lwc/pull/2263
+---
+
+# HTML comments for LWC templates
+
+## Summary
+
+In most cases, HTML comments in templates only add information for other developers. There are some use cases where HTML comments convey some runtime semantic (e.g., MSO comments). As of today, the LWC template compiler strips HTML comments. This RFC proposes a way to preserve HTML comments in LWC templates.
+
+### Basic example
+
+```html
+<template preserve-comments>
```

Comment:
Would it make sense to namespace this, just to make it clear this is an LWC-ism? (E.g. `<template lwc:preserve-comments>`.) `<template>` is being extended with things like declarative shadow DOM already (`<template shadowroot="open">`).

Reply:
ihmo, no. `preserve-comments` only applies to the root template tag, which is owned by LWC. we usually namespace with `lwc` to attributes that apply to regular elements, like `lwc:dom`.

Note: I'm not aware of `<template shadowroot="open">`, but ill assume that is meant to be used by lwc; in that case, I would say: if the same root template tag has `preserve-comments`, it should also render the comments nodes.

Reply:
Sorry for not being clear: `<template shadowroot="open">` comes from [declarative shadow DOM](https://web.dev/declarative-shadow-dom/). I was just pointing out that the browser is adding meaningful attributes to the `<template>` tag.

If LWC owns the root `<template>` then I agree it's not a big deal not to namespace it.

Reply:
I don't have a strong opinion on this, however, I like the fact that we attempt to scope all the LWC-ism with the `lwc:` prefix. This defines a clear boundary between standard and LWC specific syntax from the developer's perspective.

I am fine either way.

Reply:
Are there any cases where de _don't_ scope the attribute for LWC-isms with `lwc:`? The only ones I can think of are `for:each` and `for:item`, but those are already "scoped" with the `for`, which makes them look very non-web-standards-y since there are few standard cases that web developers will run into like that (`xmlns:xlink` on SVGs is about the only common one I can think of).

Reply:
thanks for sharing that interesting read @nolanlawson! With that, I'm in the same position as @pmdartus, I don't have a strong opinion. However, I think is safer that we go with the `lwc:`. I will update this RFC accordingly 👍 .

> Are there any cases where de don't scope the attribute for LWC-isms with `lwc:`?
on top of those: `if:true|false`, `key`, `iterator:it`?.